mobile_verifier/src/heartbeats/last_location.rs

```diff
@@ -67,58 +119,197 @@ impl LocationCache {
             WHERE location_validation_timestamp IS NOT NULL
                 AND latest_timestamp >= $1
                 AND hotspot_key = $2
+                AND $3 - location_validation_timestamp <= INTERVAL '24 hours'
             ORDER BY latest_timestamp DESC
             LIMIT 1
             "#,
         )
-        .bind(Utc::now() - Duration::hours(12))
+        .bind(Utc::now() - Duration::hours(24))
         .bind(hotspot)
+        .bind(heartbeat_timestamp)
         .fetch_optional(&self.pool)
         .await?;
-        self.locations
-            .insert(
-                hotspot.clone(),
-                last_location,
-                last_location
-                    .map(|x| x.duration_to_expiration())
-                    .unwrap_or_else(|| Duration::days(365))
-                    .to_std()?,
-            )
-            .await;
+
         Ok(last_location)
     }
 
-    pub async fn fetch_last_location(
+    async fn maybe_cache_last_location(
```

Comment:
why does it have `maybe` in the name when it always caches a value for the hotspot?